vignettes/intro_to_allotalot.Rmd

```diff
@@ -16,159 +16,165 @@ library(allotalot)
+### Unpacking a returned value
+A common use-case is when a function returns a list of values
+and you want to extract the individual values.
+In this example, the list of values returned by `coordsList()` is unpacked
+into the variables `lat` and `lng`.
+
+```{r}
+#
+# A function which returns a list of 2 numeric values.
+# 
+coordsList <- function() {
   list(38.061944, -122.643889)
 }
 
-{lat: lng} %<-% coords()
+{ lat : lng } %<-% coordsList()
 lat
 lng
 ```
 
-Notice the use of the colon and braces on the left-hand side of the assignment expression. Based on the left-hand side structure, the assignment operator expects `coords()` to return a list of two values. Those two values are then assigned to `lat` and `lng`.
+As in the earlier example,
+colon and braces on the left-hand side of the assignment
+tell the assignment operator to expect a list.
 
-Let's try rerunning the `coords` example, but this time without the braces.
+In this example, we call a function that returns a vector.
+Now we omit the braces.
 
-```{R, include = FALSE} 
-rm(lat)
-rm(lng)
-```
+```{r}
+#
+# Convert cartesian coordinates to polar
+#
+toPolar = function(x, y) {
+  c(sqrt(x^2 + y^2), atan(y/x))
+}
 
-```{R, error = TRUE}
-lat: lng %<-% coords()
-lat
-lng
+radius : angle %<-% toPolar(12, 5)
+radius
+angle
 ```
 
-To unpack a list of values, the left-hand side must be enclosed in braces. Without the braces the assignment operator expects and tries to unpack a vector of values. If we create a new function `coords2` which returns a vector, the error is fixed.
+### Example: Intercept and slope of regression
+We can directly unpack the coefficients of a simple linear regression
+into the intercept and slope.
 
-```{R}
-#' Get Another Pair of Coordinates
-#' 
-#' A function which returns a numeric vector of length 2, the coordinates.
-#'
-coords2 <- function() {
-  c(-36.85, 174.77)
-}
+```{r}
+inter : slope %<-% coef(lm(mpg ~ cyl, data = mtcars))
+inter
+slope
+```
 
-lat: lng %<-% coords2()
-lat
-lng
+### Example: Unpacking the result of `safely`
+The *purrr* package includes the `safely` function.
+It wraps a given function to create a new, "safe" version of the original function.
+
+```{R, eval = require('purrr')}
+safe_log <- purrr::safely(log)
 ```
 
-### One value to many names
+The safe version returns a list of two items.
+The first item is the result of calling the original function,
+assuming no error occurred;
+or `NULL` if an error did occur.
+The second item is the error (if an error occurred) or `NULL` (if not).
+Whether or not the original function would have thrown an error, the safe version will never throw an error.
+
+```{r, eval = require('purrr')}
+pair <- safe_log(10)
+pair$result
+pair$error
+```
 
-When unpacking a list of values we can more easily correspond names to values. However, during parallel assignment `%<-%` will try to unpack a single object on the right-hand side if multiple names are found on the left-hand side. 
+```{r}
```

Comment:
line needs to include `eval = require('purrr')` as `safe_log` will only exist if `require('purrr')` is true.